### PR TITLE
Add COLD French Law Announcement

### PR DESCRIPTION
**Pull request by clare-stanton**
We shouldn't merge this until Matteo gives the thumbs up that everything is public :)

I added Ben as a reviewer because of the GitHub gist embedding situation 
I added Matteo as a review because of the thing where we shouldn't merge until he gives the ok

### CODE REVIEW

File: app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+The final prompt that was engineered by our colleagues is shared below.
+
+## The Prompt
+
+<script src="https://gist.github.com/clare-stanton/441d8183d989c88e42def1fb2c392a3f.js"></script>
+
+---
+## Links
+[COLD French Law dataset on Hugging Face](https://huggingface.co/datasets/harvard-lil/cold-french-law)
```
Comment (matteocargnelutti) [inline app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md:38]:
```suggestion
[COLD French Law dataset on Hugging Face](https://huggingface.co/datasets/harvard-lil/cold-french-law)

```

File: app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+[COLD French Law CLI pipeline on Github](https://github.com/harvard-lil/cold-french-law-pipeline)
```
Comment (matteocargnelutti) [inline app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md:40]:
```suggestion
[COLD French Law CLI pipeline on Github](https://github.com/harvard-lil/cold-french-law-pipeline)

```

File: app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+[Other COLD data ](https://huggingface.co/datasets/harvard-lil/cold-cases)
```
Comment (matteocargnelutti) [inline app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md]:
```suggestion
See also: [COLD Cases Dataset](https://huggingface.co/datasets/harvard-lil/cold-cases)

```

File: app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Via GPT-4’s function calling feature, the pipeline was able to translate the full entries, and allowed for each relevant column of an entry to be translated in a single call (or couple of calls in the limited cases where entries were longer than 2,500 tokens.) This saved weeks of processing. Additionally, this technique outputs individual JSON files for each of the law articles.
+
+With this approach, we were able to run the pipeline for just a few hours each night, and the structure of the dataset remained intact.
+
+Over the course of this process adjustments were made to the prompt based on the expertise of the CoCounsel team and feedback provided by Timothée Charmeil, an [LL.M. candidate at HLS](https://hls.harvard.edu/graduate-program/ll-m-program/), who quality tested samples of the initial outputs.
```
Comment (matteocargnelutti) [inline app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md]:
```suggestion
Over the course of this process adjustments were made to the prompt based on the expertise of the CoCounsel team and feedback provided by **Timothée Charmeil**, an [LL.M. candidate at HLS](https://hls.harvard.edu/graduate-program/ll-m-program/), who quality tested samples of the initial outputs.
```

**Review by matteocargnelutti — APPROVED** [review]
Made a few minor edits, besides this: LGTM. Please let me know when to flip the switch on the associated repos. Thanks a million @clare-stanton for all this, stellar work.

File: app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+Via GPT-4’s function calling feature, the pipeline was able to translate the full entries, and allowed for each relevant column of an entry to be translated in a single call (or couple of calls in the limited cases where entries were longer than 2,500 tokens.) This saved weeks of processing. Additionally, this technique outputs individual JSON files for each of the law articles.
+
+With this approach, we were able to run the pipeline for just a few hours each night, and the structure of the dataset remained intact.
+
+Over the course of this process adjustments were made to the prompt based on the expertise of the CoCounsel team and feedback provided by **Timothée Charmeil**, an [LL.M. candidate at HLS](https://hls.harvard.edu/graduate-program/ll-m-program/), who quality tested samples of the initial outputs.
+
+The final prompt that was engineered by our colleagues is shared below.
```
Comment (bensteinberg) [inline app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md:30]: Maybe change from passive to active:
```suggestion
The final version of the translation prompt we used is below.
```

File: app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+Over the course of two months the CoCounsel team ran all ~800,000 articles through a translation pipeline that took each individual entry and translated it from its original French into English using OpenAI’s GPT-4 large language model. One hurdle was the variety of important metadata for each entry that was also in French, and a desire to retain each of the articles in its fullest form.
+
+Via GPT-4’s function calling feature, the pipeline was able to translate the full entries, and allowed for each relevant column of an entry to be translated in a single call (or couple of calls in the limited cases where entries were longer than 2,500 tokens.) This saved weeks of processing. Additionally, this technique outputs individual JSON files for each of the law articles.
```
Comment (bensteinberg) [inline app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md]:
```suggestion
Via GPT-4’s function-calling feature, the pipeline was able to translate the full entries, and allowed each column of an entry to be translated in a single call (or couple of calls in the limited cases where entries were longer than 2,500 tokens.) This saved weeks of processing. Additionally, this technique outputs individual JSON files for each of the law articles.
```

File: app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+This dataset—formatted into a single CSV file and [openly available on Hugging Face](https://huggingface.co/datasets/harvard-lil/cold-french-law)—contains original texts from the LEGI dataset as well as machine-generated French to English translations thanks to the participation of the CoCounsel team at [Casetext, part of Thomson Reuters](https://www.thomsonreuters.com/en/artificial-intelligence.html).
+
+COLD French Law was initially compiled to be used in a forthcoming experiment at the Lab. We are releasing it broadly today as part of our commitment to open knowledge. We see this dataset as a contribution to the quickly expanding field of legal AI, and hope it will help researchers, builders, and tinkerers of all kinds in their endeavors.
+
+---
+## The Process
+As part of these release notes we would like to share details about the process used to translate the articles contained in the dataset.
```
Comment (bensteinberg) [inline app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md]:
```suggestion
As part of these release notes, we would like to share details about the process used to translate the articles contained in the dataset.
```

File: app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+guest-author: The Library Innovation Lab Team
+tags:
+- ai
+- research
+---
+![COLD French Law banner](https://lil-blog-media.s3.amazonaws.com/COLDfrenchlaw.webp)
+
+There is a new addition to the [Collaborative Open Legal Data collection](https://huggingface.co/harvard-lil): a set of over 800,000 articles extracted from the [LEGI dataset](https://www.data.gouv.fr/fr/datasets/legi-codes-lois-et-reglements-consolides/), one of France’s official open law repositories, that were programmatically identified as “currently applicable French law” by our [pipeline](https://github.com/harvard-lil/cold-french-law-pipeline).
+
+This dataset—formatted into a single CSV file and [openly available on Hugging Face](https://huggingface.co/datasets/harvard-lil/cold-french-law)—contains original texts from the LEGI dataset as well as machine-generated French to English translations thanks to the participation of the CoCounsel team at [Casetext, part of Thomson Reuters](https://www.thomsonreuters.com/en/artificial-intelligence.html).
```
Comment (bensteinberg) [inline app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md:12]:
```suggestion
This dataset—formatted into a single CSV file and [available on Hugging Face](https://huggingface.co/datasets/harvard-lil/cold-french-law)—contains original texts from the LEGI dataset as well as machine-generated French to English translations thanks to the participation of the CoCounsel team at [Casetext, part of Thomson Reuters](https://www.thomsonreuters.com/en/artificial-intelligence.html).
```

**Review by bensteinberg — APPROVED** [review]
LGTM -- I've made a few copyediting suggestions.

File: app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md (hunk 5 above)
Comment (clare-stanton) [reply app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md:30]: We did want to highlight the fact that the prompt was engineered by the folks at CoCounsel, so going to leave as is.

File: app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md (hunk 8 above)
Comment (clare-stanton) [reply app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md:12]: It's perhaps overkill but I'm going to keep this for extra extra clarity!

File: app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+Via GPT-4’s function-calling feature, the pipeline was able to translate the full entries, and allowed each column of an entry to be translated in a single call (or couple of calls in the limited cases where entries were longer than 2,500 tokens.) This saved weeks of processing. Additionally, this technique outputs individual JSON files for each of the law articles.
+
+With this approach, we were able to run the pipeline for just a few hours each night, and the structure of the dataset remained intact.
+
+Over the course of this process adjustments were made to the prompt based on the expertise of the CoCounsel team and feedback provided by **Timothée Charmeil**, an [LL.M. candidate at HLS](https://hls.harvard.edu/graduate-program/ll-m-program/), who quality tested samples of the initial outputs.
+
+The final prompt that was engineered by our colleagues is shared below.
```
Comment (bensteinberg) [inline app/_posts/2024-05-23-announcing-the-cold-french-law-dataset.md:30]: Fair -- I'd still lose "shared" ;)
```suggestion
The final prompt that was engineered by our colleagues is below.
```

**Pull request merged**